src/CommunityToolkit.Aspire.Hosting.RavenDB/RavenDBBuilderExtensions.cs

```diff
@@ -0,0 +1,219 @@
+    /// This version of the package defaults to the <inheritdoc cref="RavenDBContainerImageTags.Tag"/> tag of the <inheritdoc cref="RavenDBContainerImageTags.Image"/> container image.
+    /// This overload simplifies configuration by accepting a <see cref="RavenDBServerSettings"/> object to specify server settings.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/></param>
+    /// <param name="name">The name of the RavenDB server resource.</param>
+    /// <param name="serverSettings">An object of type <see cref="RavenDBServerSettings"/> containing configuration details for the RavenDB server, 
+    /// such as whether the server should use HTTPS, RavenDB license and other relevant settings.</param>
+    /// <returns>A resource builder for the newly added RavenDB server resource.</returns>
+    /// <exception cref="DistributedApplicationException">Thrown when the connection string cannot be retrieved during configuration.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when the connection string is unavailable.</exception>
+    public static IResourceBuilder<RavenDBServerResource> AddRavenDB(this IDistributedApplicationBuilder builder,
+        [ResourceName] string name,
+        RavenDBServerSettings serverSettings)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var environmentVariables = GetEnvironmentVariablesFromServerSettings(serverSettings);
+        return builder.AddRavenDB(name, secured: serverSettings is RavenDBSecuredServerSettings, environmentVariables);
+    }
+
+    /// <summary>
+    /// Adds a RavenDB server resource to the application model. A container is used for local development.
+    /// This version of the package defaults to the <inheritdoc cref="RavenDBContainerImageTags.Tag"/> tag of the <inheritdoc cref="RavenDBContainerImageTags.Image"/> container image.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/></param>
+    /// <param name="name">The name of the RavenDB server resource.</param>
+    /// <param name="secured">Indicates whether the server connection should be secured (HTTPS). Defaults to false.</param>
+    /// <param name="environmentVariables">The environment variables to configure the RavenDB server.</param>
+    /// <param name="port">Optional port for the server. If not provided, defaults to the container's internal port (8080).</param>
+    /// <returns>A resource builder for the newly added RavenDB server resource.</returns>
+    /// <exception cref="DistributedApplicationException">Thrown when the connection string cannot be retrieved during configuration.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when the connection string is unavailable.</exception>
+    public static IResourceBuilder<RavenDBServerResource> AddRavenDB(this IDistributedApplicationBuilder builder,
```

Comment:
I see your point, but I think we should keep it, just in case a user wants to convert additional settings on their own or add options that aren't part of `RavenDBServerSettings`

Reply:
ok